flow/genericactors.actor.h

```diff
@@ -176,6 +176,7 @@ Future<Void> waitForAllReady( std::vector<Future<T>> results ) {
 		if (i == results.size()) return Void();
 		try {
 			T r = wait( results[i] );
+			(void)r;
```

Comment:
I think I prefer the `wait(success(...))` approach you used a few times above, since it avoids the extra warning suppression line.

Reply:
Done. I was a little concerned about adding too much indirection/extra actor allocations, but that might be cheap as far as I know. Maybe someday we can change the actor compiler to treat `success` specially and avoid allocating an intermediate actor.

Reply:
Hmm, interesting thought. Out of curiosity, I attempted a little benchmark where I waited on an already ready `Future<int>`. The results were roughly:

6ns for `int x = wait(f)`
25ns for `wait(ready(f))` that doesn't use `success`
43ns for `wait(ready(f))` using `success`

So this has a clear performance cost, but it probably only matters if you are running a lot of these. If you are, it's probably best to skip using `ready` altogether.

I'll leave it up to you which way you want to go, as I guess I don't have a strong opinion on the code cleanliness vs. performance tradeoff here.

Reply:
On GCC and clang the following might work as well (though I am not sure the actor compiler likes it):

```
T r [[gnu::unused]] = wait(results[i]);
```

We could define a macro `FLOW_UNUSED` in `actorcompiler.h` for unused and define it differently depending whether intellisense is enabled or not. So that would imho not be as ugly as `(void)r;` but much uglier than `wait(success(f));`

Reply:
I think I prefer `wait(success(...))`, and we have the option to generate more efficient code in the actor compiler someday.